### ISSUE TITLE
Duplicate dynamic imports

### ISSUE THREAD

**Issue by mrkishi** [rollup/rollup#1909]
First of all, _congratulations_ on shipping `0.55.0`! It's an awesome achievement for Rollup. 🎉 

So, while doing something stupid, I came across a bug with dynamic imports: you can't have multiple imports of the same module in a single file.

Reproduction: [Gist web view](https://gist.github.com/mrkishi/0accbb947df2b3a5b4968fb2c5baf0b4) and [Gist git clone url](https://gist.github.com/0accbb947df2b3a5b4968fb2c5baf0b4.git).

It actually works as long as you're importing external modules, but as soon as you add a local module, it fails. It also fails when `experimentalCodeSplitting` is on, although with a different stack trace (both in the gist).

---

Tangents (feel free to ignore, this is not the right place I guess):

The reason I found this out is because while testing code-splitting, I tried forcing the generation of extra bundles with:

```js
function unused() {
	Promise.all([import('./dep-a'), import('./dep-b')])
}
```

It does work, albeit funnily: `unused` is dead-code removed, but `dep-a` and `dep-b`'s bundles still get generated. I wonder if dynamic bundles should get cleaned up in these cases as well?

I know I could just add them as entries, but I was checking if webpack's `import('./some/dynamic/${paths}')` was compatible. Turns out it isn't, so, extra question: how feasible would it be for Rollup to allow plugins to trigger the generation of multiple bundles for a single import?

Again, thank you for the great release!

**Comment by guybedford** [comment]
@mrkishi thanks so much for trying this feature out and posting your findings.

I've created a fix for this issue at https://github.com/rollup/rollup/pull/1911. If you're about to test out the build, feel free, otherwise the fix should land in the next few days.

**Comment by mrkishi** [comment]
Well, that was fast. 😄 

It also seems to be working with everything I throw at it, now. Kudos!

**Comment by guybedford** [comment]
Released in 0.55.1.
